packages/schematics/angular/service-worker/files/ngsw-config.json

```diff
@@ -18,7 +18,8 @@
       "updateMode": "prefetch",
       "resources": {
         "files": [
-          "/assets/**"
+          "/assets/**",
+          "<%= resourcesOutputPath %>/*.(eot|svg|cur|jpg|png|webp|gif|otf|ttf|woff|woff2|ani)"
```

Comment:
@gkalpak this is the line that interests you.

Reply:
Note that while this covers the vast majority of cases, it is not technically complete as any file type can be used within a CSS `url()`.

The config would need to be modified at build time to account for all the resources.

Reply:
This would also break if the `resourcesOutputPath` was changed after service worker addition.  Some documentation should cover that for now, however.

Reply:
That is true, there is an alternative, but I felt it might be to inclusive. 
It's to use: `<%= resourcesOutputPath %>/*`

Reply:
@clydin, indeed that is correct `resourcesOutputPath`. I'll open a PR for AIO to document this.

That said, there is an alternative. instead of that glob, use `"<%= resourcesOutputPath %>/*` but it feels a bit to inclusive.

Reply:
> Feels a bit of a magic, and might not be clear to the users, why certain files are being included even if not whitelisted in the config.

Well, that's not much more magic than renaming and moving files around :grin:
The ideal fix, imo, would be to update all asset group configs to match the renamed/moved files that they would have matched if the files were not renamed/moved. This would essentially ensure that the cli makes up for the renaming/moving magic.

But this will be a little more complicated, so not sure if it is worth the extra effort (and costs that come with the more complicated processing).

Reply:
Thinking about it, there is no need to modify the config at build time, as this can be done in a glob like `**/*` which will catch anything which is not caught in the app group. Alternatively not to be so inclusive `**/*.(eot|svg|cur|jpg|png|webp|gif|otf|ttf|woff|woff2|ani)"` which should solve the issue of the resourceOutPath.

Also, updating the asset group configs during build time to match the actual files processed by the CLI, is pretty straight forward.

I am more inclined towards having it hardcoded in the config file for 2 main reason, experienced users have more control over the assets groups, as you cannot use negations.

1) Option one is to leave it as is in the current PR.
2) Use wildcards in the config `**/*`
2) Use wildcards with extensions  the config `**/*.(eot|svg|cur|jpg|png|webp|gif|otf|ttf|woff|woff2|ani`
3) Build assets groups during build time based on the files processed by webpack.
4) Do step 3, only if assets group `files` under assets group is empty or undefined.

@gkalpak & @clydin, what do you think?

Reply:
> Thinking about it, there is no need to modify the config at build time, as this can be done in a glob like **/* which will catch anything which is not caught in the app group.

This is not necessarily something you want to do (cache every single file in your dist directory).

> Also, updating the asset group configs during build time to match the actual files processed by the CLI, is pretty straight forward.

I am not saying it is impossible (or even too hard), but it is not as straight-forward as the current solution.
But if you feel otherwise, let's do it :smiley:

> I am more inclined towards having it hardcoded in the config file for 2 main reason, experienced users have more control over the assets groups, as you cannot use negations.

I don't understand what you are saying.
How do experienced users get more control? Why can't you use negations?

> Here are some options:
>
> Option one is to leave it as is in the current PR.

Definitely not ideal, but I can probably live with that option.

> Use wildcards in the config **/*

That doesn't sound like a good idea.

> Use wildcards with extensions the config **/*.(eot|svg|cur|jpg|png|webp|gif|otf|ttf|woff|woff2|ani

Definitely not ideal, but I can probably live with that too.

> Build assets groups during build time based on the files processed by webpack.

That would be the most accurate fix imo, but will add extra complexity/maintainance cost (for us, not for the users).
(Not convinced it is worth the extra effort, but if you feel like going that way I won't stop you :grin:)

Reply:
Negations won’t work as they are not supported by the service worker generator see: https://angular.io/guide/service-worker-config#resources

So basicilly a user will not be able to exclude a file which the CLI added via the config during build time.

I am more for the wildcard with the extension approach, it a good default and with a documentation users can configure it as they wish. 

Re the adding config during build time, it’s pretty straight forward, the only problem comes which files you want want to cache? Basically it will be the same as adding `**/*.(gif|jpg)...` in the config, since as you mentioned it doesn’t make sense to cache everything in the dist.

Reply:
How is that part of the documentation implying that you can't use negations? I am pretty sure you can :grin:
It is also mentioned near the top of the same document:

> The ! prefix marks the pattern as being negative, meaning that only files that don't match the pattern will be included.

##
Re the adding config during build time, you would need to iterate over each asset group in the config and do the following (off the top of my head):
1. Determine which files it matches (based on positive and negative glob patterns).
2. Check if any of these files have been renamed/moved.
3. Update the asset group config to also match the renamed/moved file.

But that is the super naive approach that can break with certain patterns (e.g. a negative pattern that does not match the original file but does match the renamed/moved file).

Actually, it is not possible to implement this in an 100% safe way, so probably better to go with the current approach + docs.

Reply:
I am too tired, I miss interpreted the docs 😂! Thanks George.